src/deepsparse/pipeline.py

```diff
@@ -360,7 +363,11 @@ def __call__(self, *args, **kwargs):
 
         kwargs["inference_state"] = inference_state
 
-        return self.run(*args, **kwargs)
+        next_call = self.run
+        if self.middleware_manager is not None:
+            next_call = self.middleware_manager.build_middleware_stack(next_call)
```

Comment:
this will break the async pathway since this step is missing from `run_async`?

Reply:
Are there existing tests for async ? I can add middleware there and check